hoot-core/src/main/cpp/hoot/core/algorithms/changeset/ChangesetCutOnlyCreator.cpp

```diff
@@ -395,27 +397,32 @@ void ChangesetCutOnlyCreator::create(
   const QString& input1, const QString& input2, const geos::geom::Envelope& bounds,
   const QString& output)
 {
-  LOG_VARD(input1);
-  LOG_VARD(input2);
-  LOG_VARD(GeometryUtils::envelopeToConfigString(bounds));
-  LOG_VARD(output);
+  create(input1, input2, GeometryUtils::envelopeToPolygon(bounds), output);
+}
 
+void ChangesetCutOnlyCreator::create(
+  const QString& input1, const QString& input2,
+  const std::shared_ptr<geos::geom::Polygon>& bounds, const QString& output)
+{
   QElapsedTimer timer;
   timer.start();
 
+  LOG_VARD(input1);
+  LOG_VARD(input2);
+  if (bounds)
+  {
+    LOG_VARD(GeometryUtils::polygonToString(bounds));
+  }
+  LOG_VARD(output);
+
   // INPUT VALIDATION AND SETUP
 
   _input1 = input1;
   _input1Map.reset();
   _input2 = input2;
   _input2Map.reset();
   _output = output;
-  //_replacementBounds = bounds;
-  // TODO: It makes more sense to store the bounds and then just convert it to a string as needed.
-  // The default string stores six decimal places, which should be fine for a bounds. Strangely,
-  // when I store the bounds or try to increase the precision of the bounds string, I'm getting a
-  // lot of test output issues...needs to be looked into.
-  _replacementBounds = GeometryUtils::envelopeToConfigString(bounds);
```

Comment:
Got rid of this garbage where the bounds was being stored as a string instead of an object. There were some rounding issues before, I think...gone now.